crates/pop-cli/src/commands/install/frontend.rs

```diff
@@ -0,0 +1,162 @@
+) -> Result<()> {
+	cli.info("Installing frontend development dependencies...")?;
+
+	ensure_node_v20(skip_confirm, cli).await?;
+	ensure_bun(skip_confirm, cli).await?;
+	Ok(())
+}
+
+/// Require Node v20+ is installed, and install if not present.
+///
+/// # Arguments
+/// * `skip_confirm`: If true, skip user confirmation prompts.
+/// * `cli`: Command line interface (for interactive confirm).
+pub async fn ensure_node_v20(skip_confirm: bool, cli: &mut impl cli::traits::Cli) -> Result<()> {
+	if has("node") {
+		let v = SemanticVersion::try_from("node".to_string()).map_err(|_| {
+			anyhow!("NodeJS detected but version check failed. Make sure `node --version` works.")
+		})?;
+
+		if v.0 >= MIN_NODE_VERSION {
+			return Ok(());
+		}
+	}
+	if !skip_confirm &&
+		!cli.confirm("📦 NodeJS v20+ is required. Install/upgrade now via nvm?")
+			.initial_value(true)
+			.interact()?
+	{
+		return Err(anyhow!(
+			"🚫 You have cancelled the installation process. NodeJS v20+ is required. Install from https://nodejs.org and re-run."
+		));
+	}
+	let nvm_script = install_nvm(cli).await?;
+
+	// Install node via nvm (need to source nvm first)
+	let install_cmd = format!(r#"source "{}" && nvm install node"#, nvm_script);
+	cmd("bash", vec!["-c", &install_cmd]).run()?;
+	Ok(())
+}
+
+/// Ensure Bun exists and return the absolute path to the `bun` binary.
+///
+/// # Arguments
+/// * `skip_confirm`: If true, skip user confirmation prompts.
+/// * `cli`: Command line interface.
+pub async fn ensure_bun(skip_confirm: bool, cli: &mut impl cli::traits::Cli) -> Result<PathBuf> {
+	if let Some(path) = which("bun") {
+		return Ok(PathBuf::from(path));
+	}
+	if !skip_confirm &&
+		!cli.confirm(
+			"📦 Do you want to proceed with the installation of the following package: bun ?",
+		)
+		.initial_value(true)
+		.interact()?
+	{
+		return Err(anyhow!("🚫 You have cancelled the installation process."));
+	}
+	// Install Bun using the official installer script
+	run_external_script("https://bun.sh/install", &[]).await?;
+
+	// Try to locate bun in PATH again
+	if let Some(path) = which("bun") {
+		return Ok(PathBuf::from(path));
+	}
+	// Use the default install location from the official installer if not found in PATH
+	let home = std::env::var("HOME").map_err(|_| anyhow!("HOME not set"))?;
+	let bun_abs = PathBuf::from(format!("{home}/.bun/bin/bun"));
+
+	if !bun_abs.exists() {
+		return Err(anyhow!(format!(
+			"Bun installed but not found at {}. Open a new shell or add it to PATH.",
+			bun_abs.display()
+		)));
+	}
+	Ok(bun_abs)
+}
+
+/// Require `npx` to be available.
+pub fn ensure_npx() -> Result<()> {
+	if !has("npx") && !has("npm") {
+		return Err(anyhow!(
+			"`npx` (or npm with npx) not found on PATH. Install NodeJS from https://nodejs.org and re-run."
+		));
+	}
+	Ok(())
+}
+
+fn has(bin: &str) -> bool {
+	cmd("which", vec![bin]).read().is_ok()
+}
+
+fn which(bin: &str) -> Option<String> {
+	cmd("which", vec![bin]).read().ok().map(|s| s.trim().to_string())
```

Comment:
I don't think there's a need to trim here. I definitely don't think the `which` command includes unnecessary whitespaces in its output.

Reply:
True, https://github.com/r0gue-io/pop-cli/pull/676/commits/849199d0d98ec36684454b4e1160b582c8497a9e